docs/develop/power-automate-integration.md

```diff
@@ -37,26 +37,22 @@ All script input is specified as additional parameters for the `main` function.
 
 When you're configuring a flow in Power Automate, you can specify script input as static values, [expressions](/power-automate/use-expressions-in-conditions), or dynamic content. Details on an individual service's connector can be found in the [Power Automate Connector documentation](/connectors/).
 
-When adding input parameters to a script's `main` function, consider the following allowances and restrictions.
+#### Type restrictions
 
-1. The first parameter must be of type `ExcelScript.Workbook`. Its parameter name does not matter.
+When adding input parameters to a script's `main` function, consider the following allowances and restrictions. These also apply to the return type of the script.
 
-1. Every parameter must have a type (such as `string` or `number`).
+1. The first parameter must be of type `ExcelScript.Workbook`. Its parameter name doesn't matter.
 
-1. The basic types `string`, `number`, `boolean`, `unknown`, `object`, and `undefined` are supported.
+1. The types `string`, `number`, `boolean`, `unknown`, `object`, and `undefined` are supported.
 
-1. Arrays (`[]`) of the previously listed basic types are supported.
-    > [!IMPORTANT]
-    > Please note that the object `Array<T>` is not a supported parameter type.
+1. Arrays (both `[]` and `Array<T>` styles) of the previously listed types are supported. Nested arrays are also supported.
 
-1. Nested arrays are supported as parameters (but not as return types).
+1. Union types are allowed if they are a union of literals belonging to a single type (such as `"Left" | "Right"`, not `"Left", 5`). Unions of a supported type with undefined are also supported (such as `string | undefined`).
 
-1. Union types are allowed if they are a union of literals belonging to a single type (such as `"Left" | "Right"`). Unions of a supported type with undefined are also supported (such as `string | undefined`).
-
-1. Object types are allowed if they contain properties of type `string`, `number`, `boolean`, supported arrays, or other supported objects. The following example shows nested objects that are supported as parameter types:
+1. Object types are allowed if they contain properties of type `string`, `number`, `boolean`, supported arrays, or other supported objects. The following example shows nested objects that are supported as parameter types.
 
     ```TypeScript
-    // Office Scripts can return an Employee object because Position only contains strings and numbers.
+    // Office Scripts can use an Employee object because Position is also supported.
```

Comment:
Maybe instead of `Position is also supported` say something like `Position is also supported because its properties are a supported type`

Reply:
How do you feel about `// The Employee object is supported because Position is also composed of supported types.`

Reply:
Yea that sounds good!